src/System.Memory/src/System/Buffers/ReadOnlySequence.cs

```diff
@@ -397,22 +403,42 @@ public ReadOnlySequence<T> Slice(SequencePosition start, long length)
         /// <summary>
         /// Forms a slice out of the given <see cref="ReadOnlySequence{T}"/>, beginning at <paramref name="start"/>, ending at <paramref name="end"/> (inclusive).
         /// </summary>
-        /// <param name="start">The starting (inclusive) <see cref="SequencePosition"/> at which to begin this slice.</param>
-        /// <param name="end">The ending (inclusive) <see cref="SequencePosition"/> of the slice</param>
+        /// <param name="start">The starting (inclusive) <see cref="SequencePosition"/> at which to begin this slice. If <code>default</code> the result begins at the existing <see cref="ReadOnlySequence{T}"/>'s start.</param>
+        /// <param name="end">The ending (inclusive) <see cref="SequencePosition"/> of the slice. If <code>default</code> the result ends at the existing <see cref="ReadOnlySequence{T}"/>'s end.</param>
         [MethodImpl(MethodImplOptions.AggressiveInlining)]
         public ReadOnlySequence<T> Slice(SequencePosition start, SequencePosition end)
         {
+            if (end.Equals(default))
```

Comment:
I think this code path is perf sensitive (esp since its inlined). Could we instead do the following:
```C#
if (end.Equals(default) || start.Equals(default))
{
   // call SliceNonInlinedHelper
}
```

I would imagine passing default to be the uncommon case.